src/Umbraco.Web/PublishedCache/ContentCacheAdapter.cs

```diff
@@ -0,0 +1,218 @@
+        // should get rid of hideTopLevelNode and other oddities entirely, eventually
+
+        public IPublishedContent GetByRoute(bool preview, string route, bool? hideTopLevelNode = null, string culture = null)
+        {
+            if (route == null) throw new ArgumentNullException(nameof(route));
+
+            var cache = preview == false || PublishedSnapshotService.FullCacheWhenPreviewing ? _elementsCache : _snapshotCache;
+            var key = NuCache.CacheKeys.ContentCacheContentByRoute(route, preview, culture);
+            return cache.GetCacheItem<IPublishedContent>(key, () => _contentCache.GetById(_contentRouter.GetIdByRoute(_contentCache, preview, route, hideTopLevelNode, culture).Id));
+        }
+
+        public IPublishedContent GetByRoute(string route, bool? hideTopLevelNode = null, string culture = null)
+        {
+            bool preview = _contentCache.PreviewDefault;
+            return GetByRoute(preview, route, hideTopLevelNode, culture);
+        }
+
+        public IEnumerable<IPublishedContent> GetByXPath(bool preview, string xpath, params XPathVariable[] vars)
+        {
+            return _contentCache.GetByXPath(preview, xpath, vars);
+        }
+
+        public IEnumerable<IPublishedContent> GetByXPath(string xpath, params XPathVariable[] vars)
+        {
+            return _contentCache.GetByXPath(xpath, vars);
+        }
+
+        public IEnumerable<IPublishedContent> GetByXPath(bool preview, XPathExpression xpath, params XPathVariable[] vars)
+        {
+            return _contentCache.GetByXPath(preview, xpath, vars);
+        }
+
+        public IEnumerable<IPublishedContent> GetByXPath(XPathExpression xpath, params XPathVariable[] vars)
+        {
+            return _contentCache.GetByXPath(xpath, vars);
+        }
+
+        public IPublishedContentType GetContentType(Guid key)
+        {
+            return _contentCache.GetContentType(key);
+        }
+
+        public IPublishedContentType GetContentType(int id)
+        {
+            return _contentCache.GetContentType(id);
+        }
+
+        public IPublishedContentType GetContentType(string alias)
+        {
+            return _contentCache.GetContentType(alias);
+        }
+
+        public string GetRouteById(bool preview, int contentId, string culture = null)
+        {
+            var cache = (preview == false || PublishedSnapshotService.FullCacheWhenPreviewing) ? _elementsCache : _snapshotCache;
+            var key = NuCache.CacheKeys.ContentCacheRouteByContent(contentId, preview, culture);
+            return cache.GetCacheItem<string>(key, () =>
+            {
+                var result = _contentRouter.GetRouteById(_contentCache, _domainCache, preview, contentId, culture);
+                if (result.Outcome == RoutingOutcome.NotFound)
+                {
+                    return null;
+                }
+                return (result.DomainId?.ToString(CultureInfo.InvariantCulture) ?? "") + result.Url;
```

Comment:
Guess we really need code comments/docs for what this is. We don't have them anywhere that I can see :( I'll update the current source with something. I'll make another internal note for this to clean up in netcore too since we don't need to worry about breaking changes and can hopefully just deal with RouteByIdResult instead.